File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/credit/port/CreditTransactionRepository.java
package com.study.webflux.rag.domain.credit.port;

import org.springframework.data.domain.Pageable;

import com.study.webflux.rag.domain.credit.model.CreditTransaction;
import com.study.webflux.rag.domain.dialogue.model.UserId;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface CreditTransactionRepository {

	Mono<CreditTransaction> save(CreditTransaction transaction);

	Flux<CreditTransaction> findByUserIdOrderByCreatedAtDesc(UserId userId, Pageable pageable);
}

File: backend/application/src/main/kotlin/com/miyou/app/application/credit/port/CreditTransactionRepository.kt
package com.miyou.app.application.credit.port

import com.miyou.app.domain.credit.model.CreditTransaction
import org.springframework.data.domain.Pageable
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface CreditTransactionRepository {
    fun save(transaction: CreditTransaction): Mono<CreditTransaction>

    fun findByUserIdOrderByCreatedAtDesc(
        userId: String,
        pageable: Pageable,
    ): Flux<CreditTransaction>
}
